### PR TITLE
net:rpl: Null pointer dereferences

### LINKED ISSUE #10094: [Coverity CID :188169] Null pointer dereferences in /subsys/net/ip/rpl.c
Static code scan issues seen in File: /subsys/net/ip/rpl.c
 Category: Null pointer dereferences
 Function: dio_timer
 Component: Networking
 CID: 188169
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by satyacube**
Check for NULL IPv6 addr values from net_if_ipv6_get_ll() in rpl.c
If NULL, print a debug statement
Coverity-CID: 188169

Fixes #10094.

Signed-off-by: Satya Bhattacharya <satyacube@gmail.com>

### CODE REVIEW

File: subsys/net/ip/rpl.c [hunk 1]
```diff
@@ -649,13 +649,20 @@ static void dio_timer(struct k_work *work)
 			struct in6_addr *addr =
 				net_if_ipv6_get_ll(rpl_default_iface,
 						   NET_ADDR_PREFERRED);
+			if (addr) {
+				net_rpl_dio_send(rpl_default_iface, instance, addr,
+					         NULL);
+				#if defined(CONFIG_NET_STATISTICS_RPL)
```
Comment (jukkar) [inline subsys/net/ip/rpl.c]: We typically do not use indented pre-processor code, so #if should start from first column.

**Review by jukkar — CHANGES_REQUESTED** [review]
Looks ok, minor tweak needed. See also https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/24243/5/tests about the errors the sanitychecker is reporting.

**Review by pfalcon — CHANGES_REQUESTED** [review]
Commit message doesn't correspond to the change made. The commit message says:

> Check for NULL IPv6 addr values from net_if_ipv6_get_ll() in rpl.c
If NULL, print a debug statement

But that doesn't describe what you're doing fully. Because besides printing debug statement, you actually schedule a delayed work. IMHO, the commit message should explain why you're doing that (I for one can't assess if that's a right change or not).

**Comment by pfalcon** [comment]
recheck

**Comment by pfalcon** [comment]
@satyacube:

Thanks for the updates!

> Do you guys know how to trigger the Shippable sanity runs?

On each new push it should trigger itself, and if something goes wrong, you can manually trigger it by posting comment of just "recheck" (like I did above).

**Review by pfalcon — DISMISSED** [review]
Thanks for the updates! Let me approve from my side, but I hope people more knowledge in RPL will eyeball it too.

File: subsys/net/ip/rpl.c [hunk 2]
```diff
@@ -650,12 +650,20 @@ static void dio_timer(struct k_work *work)
...
 
-			net_rpl_dio_send(rpl_default_iface, instance, addr,
-					 NULL);
+			if (addr) {
+				net_rpl_dio_send(rpl_default_iface, instance,
+						 addr, NULL);
+				#if defined(CONFIG_NET_STATISTICS_RPL)
+				instance->dio_send_pkt++;
+				#endif
+			} else {
```
Comment (jfischer-phytec-iot) [inline subsys/net/ip/rpl.c]: The case (addr == NULL) will only occur if, for example, an interface(802.15.4) do not exist or something went wrong when initializing the driver. I do not know how much sense it makes to resubmit here. See also the else branch below (L667). @rveerama1 Can you comment on that?

Comment (rveerama1) [reply subsys/net/ip/rpl.c]: I agree with @jfischer-phytec-iot. This doesn't make sense to resubmit after timeout. You end up with same addr is NULL. It's better to through NET_ERR message and find the root cause of addr being NULL. Do you have any test case or sample app with this scenario other than coverity issue?

**Review by rveerama1 — CHANGES_REQUESTED** [review]

File: subsys/net/ip/rpl.c (hunk 2 above)
Comment (jfischer-phytec-iot) [reply subsys/net/ip/rpl.c]: I observed it when I did not plug in FRDM-CR20A properly while testing rpl_border_router. To provoke it, it is enough to try rpl_border_router or rpl_node on FRDM-K64F without a FRDM-CR20A.

File: subsys/net/ip/rpl.c [hunk 3]
```diff
@@ -650,12 +650,18 @@ static void dio_timer(struct k_work *work)
...
+				NET_ERR("Cannot send DIO, IPv6 link local "
+					"address is NULL");
+			return;
```
Comment (jukkar) [inline subsys/net/ip/rpl.c]: Indentation wrong here

File: subsys/net/ip/rpl.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/ip/rpl.c]: Please start the #if from the column 0

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by jukkar** [comment]
BTW, there is no need to constantly close/re-open this PR. If you have a new version, just force push it over the old thread. The PR will be closed automatically when it is merged to upstream.

File: subsys/net/ip/rpl.c [hunk 4]
```diff
@@ -650,12 +650,18 @@ static void dio_timer(struct k_work *work)
...
 
-			net_rpl_dio_send(rpl_default_iface, instance, addr,
-					 NULL);
-
+			if (addr) {
+				net_rpl_dio_send(rpl_default_iface, instance,
+						 addr, NULL);
 #if defined(CONFIG_NET_STATISTICS_RPL)
-			instance->dio_send_pkt++;
+	instance->dio_send_pkt++;
```
Comment (jukkar) [inline subsys/net/ip/rpl.c]: The indentation of this line is wrong, it should be placed in same column as the net_rpl_dio_send() above.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: subsys/net/ip/rpl.c (hunk 4 above)
Comment (satyacube) [reply subsys/net/ip/rpl.c]: Ah! I missed that one. Thanks!

**Review by jukkar — APPROVED** [review]

**Review by rveerama1 — APPROVED** [review]

**Pull request merged**